pkg/hubble/observer/local_observer.go

```diff
@@ -306,6 +333,128 @@ nextEvent:
 	}
 }
 
+// GetAgentEvents implements observerpb.ObserverClient.GetAgentEvents.
+func (s *LocalObserverServer) GetAgentEvents(
+	req *observerpb.GetAgentEventsRequest,
+	server observerpb.Observer_GetAgentEventsServer,
+) (err error) {
+	ctx, cancel := context.WithCancel(server.Context())
+	defer cancel()
+
+	var whitelist, blacklist filters.FilterFuncs
+
+	start := time.Now()
+	log := s.GetLogger()
+	ring := s.GetRingBuffer()
+
+	i := uint64(0)
+	defer func() {
+		log.WithFields(logrus.Fields{
+			"number_of_agent_events": i,
+			"buffer_size":            ring.Cap(),
+			"took":                   time.Since(start),
+		}).Debug("GetAgentEvents finished")
+	}()
+
+	ringReader, err := newRingReader(ring, req, whitelist, blacklist)
+	if err != nil {
+		if err == io.EOF {
```

Comment:
nitpicking:
```suggestion
		if errors.Is(err, io.EOF) {
```
Could do the same below and in `GetFlows`.